Mathlib/AlgebraicTopology/SingularHomology.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Algebra.Category.ModuleCat.Abelian
+import Mathlib.Algebra.Category.ModuleCat.Adjunctions
+import Mathlib.AlgebraicTopology.AlternatingFaceMapComplex
+import Mathlib.AlgebraicTopology.SingularSet
+
+/-!
+# Singular homology
+
+In this file, we define the singular chain complex and singular homology of a topological space.
+We also calculate the homology of a totally disconnected space as an example.
+
+-/
+
+noncomputable section
+
+namespace AlgebraicTopology
+
+open CategoryTheory
+
+variable (R : Type) [CommRing R] (n : ℕ)
+
+/-- The singular chain complex functor with coefficients in `R`. -/
+def singularChainComplexFunctor : TopCat.{0} ⥤ ChainComplex (ModuleCat R) ℕ :=
```

Comment:
I would think this should be defined first as a functor from the category of simplicial sets, and secondly from topological spaces. Also, it would be nice to have a more general definition for the "coefficients". At the very least, the chain complex with coefficient `M` should be defined for any `M : ModuleCat R` (and not only for `R` itself).
One approach would be to take a preadditive "coefficient category `C`" and an object `X : C`, and consider the extension to simplicial objects of the functor which sends a type `A` to the coproduct of copies of `X` indexed by `A` (assume `C` has coproducts of size `u` for a certain universe `u`).

Reply:
It would be reasonable to move this to `AlgebraicTopology.SingularHomology.Basic`.